plugins/cilium-cni/chaining/api/api.go

```diff
@@ -85,10 +88,18 @@ func Register(name string, p ChainingPlugin) error {
 	return nil
 }
 
-// Lookup searches for a chaining plugin with a given name and returns it
+// Lookup searches for a chaining plugin with a given name and returns it.
+// If no plugin with that name exists, returns the generic veth plugin.
 func Lookup(name string) ChainingPlugin {
 	mutex.RLock()
 	defer mutex.RUnlock()
+	if name == DefaultConfigName {
+		return nil
+	}
+
+	if plugin, exists := chainingPlugins[name]; exists {
+		return plugin
+	}
+	return chainingPlugins[GenericConfigName]
```

Comment:
I think we need to add an option for this to not surprise users as we keep adding new chaining modes. How about we add:
```
type NetConf struct {
        cniTypes.NetConf
        chainingMode string
        [...]
```

So you can simply define the chaining mode in the CNI configuration file? If that option is set, the name is ignored and the chaining mode is used and it fails hard if the chaining mode is unknown.